diff --git a/index.d.ts b/index.d.ts
index fab37fd606a62cd2dc81ff0b8007a26447916438..f020700e65e7de0ba58c51a08eea54da505b07a9 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -81,7 +81,7 @@ declare class EventEmitter<
 
 declare namespace EventEmitter {
   export interface ListenerFn<Args extends any[] = any[]> {
-    (...args: Args): void;
+    (...args: Args): any;
   }
 
   export interface EventEmitterStatic {
@@ -96,7 +96,7 @@ declare namespace EventEmitter {
    * ```
    * interface EventTypes {
    *   'event-with-parameters': any[]
-   *   'event-with-example-handler': (...args: any[]) => void
+   *   'event-with-example-handler': (...args: any[]) => any
    * }
    * ```
    */
@@ -107,7 +107,7 @@ declare namespace EventEmitter {
     : keyof T;
 
   export type ArgumentMap<T extends object> = {
-    [K in keyof T]: T[K] extends (...args: any[]) => void
+    [K in keyof T]: T[K] extends (...args: any[]) => any
       ? Parameters<T[K]>
       : T[K] extends any[]
       ? T[K]
@@ -118,10 +118,10 @@ declare namespace EventEmitter {
     T extends ValidEventTypes,
     K extends EventNames<T>
   > = T extends string | symbol
-    ? (...args: any[]) => void
+    ? (...args: any[]) => any
     : (
         ...args: ArgumentMap<Exclude<T, string | symbol>>[Extract<K, keyof T>]
-      ) => void;
+      ) => any;
 
   export type EventArgs<
     T extends ValidEventTypes,
diff --git a/node_modules/.vitest/deps/_metadata.json b/node_modules/.vitest/deps/_metadata.json
new file mode 100644
index 0000000000000000000000000000000000000000..2fd1ba52f100dc9c1f021bd113efc6619f56ce18
--- /dev/null
+++ b/node_modules/.vitest/deps/_metadata.json
@@ -0,0 +1,8 @@
+{
+  "hash": "23f13a44",
+  "configHash": "8c6e2c97",
+  "lockfileHash": "e3b0c442",
+  "browserHash": "6a74185e",
+  "optimized": {},
+  "chunks": {}
+}
\ No newline at end of file
diff --git a/node_modules/.vitest/deps/package.json b/node_modules/.vitest/deps/package.json
new file mode 100644
index 0000000000000000000000000000000000000000..3dbc1ca591c0557e35b6004aeba250e6a70b56e3
--- /dev/null
+++ b/node_modules/.vitest/deps/package.json
@@ -0,0 +1,3 @@
+{
+  "type": "module"
+}
